src/main/webapp/js/community-page-loader.js

```diff
@@ -39,16 +44,46 @@
 		
         if(aboutMe)
         {
-          const aboutMeText = document.createElement('textarea');
-          aboutMeText.setAttribute('readonly', 'readonly');
-          aboutMeText.innerHTML = aboutMe;
-          userListItem.appendChild(aboutMeText);
+          const aboutMeDiv = document.createElement('Div');
+          aboutMeDiv.innerHTML = aboutMe;
+          userListItem.appendChild(aboutMeDiv);
         }
         return userListItem;
       }
+	  
+	  /*filers users by key word in their name or aboutMe description*/
+	  function filterUsers() {
+		const keyword = document.getElementById('search-text').value;
+		const listul = document.getElementById('list');
+        listul.innerHTML = '';
+		var filteredList = [];
+		
+		if(userNameAndAboutList.length!=0)
+		{
+           if(keyword=="" || keyword===undefined)
+		   {
+             fetchUserList();
+		   }
+		   else
+		   {
+             /*make a list of filtered users*/
+             for(var i = 0; i<userNameAndAboutList.length; i++) {
+               if(userNameAndAboutList[i].toLowerCase().includes(keyword.toLowerCase()))
```

Comment:
are you trying to search for users ? I think it'll be better to use the datastore search functionality instead because right now you are sending the entire list of users each time.

Reply:
It does n't send the list of users each time, and it does use Datastore (in the fetch function that populates userNameAndAboutList array) . So what it does is that it fetches a whole list of charities from Datastore, but stores it in an array (userNameAndAboutList). so that if someone searches for a name, it can search the array and find that record rather than going all the way back to server again to extract the same data (it seems like a lot of overhead for the same information)

Reply:
Just note that this approach might lead to stale data if if the user stays in the page for a long time, if it only queries the database once (on load). You might want to add a refresh button and a timestamp to let the user know that they might be viewing old data.

Reply:
It doesn't fetch just once, but every time the search area is empty is fetched data from datastore, or every time page is refreshed it fetches data from data store. We can easily change the code to fetch the datastore every time the users searches a keyword, but it seems unnecessary to query datastore every time for part of data that we already have. That's a choice to make: Do we want to query datastore for search everytime for any type of search? or do we want to fetch the datastore for all records search, and reuse that data for partial search?

Reply:
Sounds good.